zdm/iteration.py

```diff
@@ -28,35 +29,49 @@ def get_likelihood(pset,grid,survey,norm=True,psnr=False):
 		ng=1
 		ns=1
 	if ng==1:
-		update_grid(grid,pset)
+		update_grid(grid,pset,survey)
 		if survey.nD==1:
-			loglik=calc_likelihoods_1D(grid,survey,pset,norm=norm,psnr=psnr)
+			#loglik=calc_likelihoods_1D(grid,survey,pset,norm=norm,psnr=psnr)
+			llsum,lllist,expected=calc_likelihoods_1D(grid,survey,pset,norm=norm,psnr=True,dolist=1)
 		elif survey.nD==2:
-			loglik=calc_likelihoods_2D(grid,survey,pset,norm=norm,psnr=psnr)
-		
+			#loglik=calc_likelihoods_2D(grid,survey,pset,norm=norm,psnr=psnr)
+			llsum,lllist,expected=calc_likelihoods_2D(grid,survey,pset,norm=norm,psnr=True,dolist=1)
+		return llsum,lllist,expected
+		#negative loglikelihood is NOT returned, positive is.	
 	else:
 		loglik=0
 		for i,g in enumerate(grid):
 			s=survey[i]
-			update_grid(g,pset)
+			update_grid(g,pset,s)
```

Comment:
Out of interest, what information does update_grid need from this survey?

Reply:
Along with the grid and pset, because we are recalculating thresholds the `update_grid` function requires the survey file as well because `survey.meta['THRESH'], survey.efficiencies and survey.wplist` are required for `calc_thresholds`